lib/async_hooks.js

```diff
@@ -220,10 +220,11 @@ class AsyncResource {
     return this[trigger_async_id_symbol];
   }
 
-  bind(fn) {
+  bind(fn, thisArg) {
     if (typeof fn !== 'function')
       throw new ERR_INVALID_ARG_TYPE('fn', 'Function', fn);
-    const ret = FunctionPrototypeBind(this.runInAsyncScope, this, fn);
+    const ret =
+      FunctionPrototypeBind(this.runInAsyncScope, this, fn, thisArg || this);
```

Comment:
if `thisArg` is falsy (e.g. false, 0,...) `this` will be used instead the passed `thisArg`.
Maybe use a check on arguments count passed instead?